pkg/didcomm/protocol/didexchange/service.go

```diff
@@ -422,84 +426,19 @@ func (s *Service) AcceptExchangeRequest(connectionID, publicDID, label string) e
 func (s *Service) RespondTo(i *OOBInvitation) (string, error) {
 	i.Type = oobMsgType
 
-	record, err := s.createConnectionRecord(i)
-	if err != nil {
-		return "", fmt.Errorf("failed to create connection record for oobinvitation : %w", err)
-	}
-
-	msg := &message{
-		Msg:           service.NewDIDCommMsgMap(i),
-		ThreadID:      i.ThreadID,
-		Options:       &options{},
-		NextStateName: stateNameInvited,
-		ConnRecord:    record,
-	}
-
-	err = s.storeEventTransientData(msg)
-	if err != nil {
-		return "", fmt.Errorf("failed to store transient event data : %w", err)
-	}
-
-	err = s.AcceptInvitation(record.ConnectionID, record.InvitationDID, record.TheirLabel)
-	if err != nil {
-		return "", fmt.Errorf("failed to accept oobinvitation : %w", err)
-	}
-
-	return record.ConnectionID, nil
+	return s.HandleInbound(service.NewDIDCommMsgMap(i), "", "")
 }
 
-func (s *Service) createConnectionRecord(i *OOBInvitation) (*connection.Record, error) {
-	if i.ThreadID == "" {
-		return nil, errors.New("missing thread ID in invitation")
-	}
-
-	if i.Target == nil {
-		return nil, errors.New("missing target in invitation")
-	}
-
-	record := &connection.Record{
-		ConnectionID: generateRandomID(),
-		ThreadID:     i.ThreadID,
-		State:        stateNameInvited, // TODO
-		InvitationID: i.ID,
-		TheirLabel:   i.Label,
-		Namespace:    myNSPrefix, // TODO
-	}
-
-	var svc *did.Service
-
-	switch target := i.Target.(type) {
-	case string:
-		doc, err := s.ctx.vdriRegistry.Resolve(target)
-		if err != nil {
-			return nil, fmt.Errorf("failed to resolve did=%s : %w", target, err)
-		}
-
-		var ok bool
-
-		svc, ok = did.LookupService(doc, didCommServiceType)
-		if !ok {
-			return nil, fmt.Errorf("no service type=%s found for did doc=%+v", didCommServiceType, doc)
-		}
-
-		record.InvitationDID = doc.ID
-		record.Implicit = true
-	case *did.Service:
-		svc = target
-	default:
-		return nil, fmt.Errorf("unsupported target type: %+v", target)
-	}
-
-	record.RecipientKeys = svc.RecipientKeys
-	record.RoutingKeys = svc.RoutingKeys
-	record.ServiceEndPoint = svc.ServiceEndpoint
+// Save this invitation created by you.
+func (s *Service) Save(i *OOBInvitation) error {
```

Comment:
I feel better to have a suffix in the name - SaveInvitation or SaveOOBInvitation or something similar.

Reply:
Yeh - that would seem more clear.

Reply:
I was led to believe we did that because golang doesn't feature method overloading.

Reply:
Also, it would be inconsistent when someone else adds a new SaveXYZ API. We would end up with Save() and SaveXYZ() functions.

Reply:
Ok - I disagree, but I'm not dying on this hill. I renamed it to "SaveInvitation".

Here are my own thoughts on the matter:

* There is a TODO to rename `OOBInvitation` - I'm not even sure it'll end up being called "Invitation".
* People worrying about refactoring a method's name makes me worry about the fragility of the code and the process.
* There seems to be a strong desire to attempt to predict the future. Let's just embrace change (the agile way) :). I don't see this service saving anything else, but that is my own perception of things, and it's limited to *today*
* There is a tendency for overly verbose method and variable names that encumbers the code's legibility